pkg/metrics/metrics.go

```diff
@@ -185,6 +184,37 @@ func (pc *PrometheusNodeConfig) Configure() error {
 	return nil
 }
 
+// retrieveMetricsEndpoint returns an existing Endpoint object or creates a new one in the operator namespace
+// if it does not already exist.
+func (pc *PrometheusNodeConfig) retrieveMetricsEndpoint() (*v1.Endpoints, error) {
+	existingEndpoint, err := pc.k8sclientset.CoreV1().Endpoints(pc.namespace).Get(context.TODO(), windowsMetricsResource, metav1.GetOptions{})
+	if err == nil {
+		return existingEndpoint, nil
+	}
+	// create new Endpoint
+	newEndpoint := &v1.Endpoints{
+		TypeMeta: metav1.TypeMeta{
+			Kind: "Endpoints",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      windowsMetricsResource,
+			Namespace: pc.namespace,
+			Labels:    map[string]string{"app.kubernetes.io/name": windowsMetricsResource},
+		},
+		Subsets: nil,
+	}
+
+	if apierrors.IsNotFound(err) {
+		// Create Endpoint object since it does not exist
+		endpoint, err := pc.k8sclientset.CoreV1().Endpoints(pc.namespace).Create(context.TODO(), newEndpoint, metav1.CreateOptions{})
+		if err != nil {
+			return nil, errors.Wrap(err, "error creating Metrics Endpoint")
```

Comment:
This can be better organized to reduce indentation and return early on error:

```

	existingEndpoint, err := pc.k8sclientset.CoreV1().Endpoints(pc.namespace).Get(context.TODO(), windowsMetricsResource, metav1.GetOptions{})
	if err == nil {
		return existingEndpoint, nil
	} else if err != apierrors.IsNotFound(err) {
		return nil, errors.Wrap(err, "could not retrieve Metrics Endpoint")
	}
	<... rest of code ...>

```

Reply:
updated to separate out.